frontend/catalyst/pennylane_extensions.py

```diff
@@ -428,7 +432,78 @@ def circuit(x):
     >>> workflow(2.0)
     array(-3.14159265)
     """
-    return Grad(_ensure_differentiable(f), grad_params=_check_grad_params(method, h, argnum))
+    scalar_out = True
+    return Grad(
+        _ensure_differentiable(f), grad_params=_check_grad_params(method, scalar_out, h, argnum)
+    )
+
+
+def jacobian(f: DifferentiableLike, *, method=None, h=None, argnum=None):
+    """A :func:`~.qjit` compatible Jacobian transformation for PennyLane/Catalyst.
+
+    This function allows the Jacobian of a hybrid quantum-classical function
+    to be computed within the compiled program.
+
+    .. warning::
+
+        Currently, higher-order differentiation or differentiation of non-QNode functions
+        is only supported by the finite-difference method.
+
+    .. note::
+
+        Any JAX-compatible optimization library, such as `JAXopt
+        <https://jaxopt.github.io/stable/index.html>`_, can be used
+        alongside ``jacobian`` for JIT-compatible variational workflows.
+        See the :doc:`/dev/quick_start` for examples.
+
+    Args:
+        f (Callable): a function or a function object to differentiate
+        method (str): The method used for differentiation, which can be any of
+    ``["fd", "defer"]``, where:
```

Comment:
The indent here seems off

Reply:
Oops just Joshes comment on this. Note that grad still has this issue as well.

Reply:
Fixed: https://github.com/PennyLaneAI/catalyst/pull/254/commits/ce73355bd623f930fe814bfb67e568b0b64e14ab

Reply:
Apologies, I think the docstring Josh suggested still doesn't render right :
![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/2117d32a-3c94-4846-baa3-5efe3e9ac161)

The best outcome I've been able to get is this, which is at least consistent in the string form and rendered form but does introduce a new line before the description:
![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/30821054-31aa-4a8a-920b-ca5108806cb8)

Produced by the following:

```py
        method (str): The method used for differentiation, which can be any of ``["fd", "defer"]``,
                      where:

                      - ``"fd"`` represents first-order finite-differences for the entire hybrid
                        circuit,

                      - ``"defer"`` represents deferring the quantum differentiation to the method
                        specified by the QNode, while the classical computation is differentiated
                        using traditional auto-diff.
```

Reply:
thanks @dime10! this is what I was trying to aim for but missed 🤦

Reply:
Alright, the docs build appears to look like this as of https://github.com/PennyLaneAI/catalyst/pull/254/commits/443219679ac5554b74161f9f8c13ebe9252c6bab